paimon-core/src/main/java/org/apache/paimon/mergetree/LookupFile.java

```diff
@@ -93,7 +126,7 @@ public static Cache<String, LookupFile> createCache(
     }
 
     private static int fileWeigh(String file, LookupFile lookupFile) {
-        return fileKibiBytes(lookupFile.localFile);
+        return lookupFile.isUsed() ? 0 : fileKibiBytes(lookupFile.localFile);
```

Comment:
Marking the boolean is to make the weight to zero while this file is using. This could avoid the lookup file being evicted when putting it to a nearly full cache.

Reply:
Can we just:
```
LookupFile lookupFile = lookupFileCache.getIfPresent(file.fileName());
boolean newLookupFile = false;
if (lookupFile is null or lookupFile is closed) {
     lookupFile = createLookupFile();
     newLookupFile = true;
}
try {
     ..... use lookup file
} finally {
    if (newLookupFile) {
           lookupFileCache.put(file.fileName(), lookupFile);
    }
}


```

Reply:
`lookupFileCache.put(file.fileName(), newLookupFile);` This still could evict this new create file out directly?